packages/amplify-graphql-model-transformer/src/resources/rds-model-resource-generator.ts

```diff
@@ -31,16 +28,33 @@ export class RdsModelResourceGenerator extends ModelResourceGenerator {
       const engine = getEngineFromDBType(dbType);
       const secretEntry = context.datasourceSecretParameterLocations.get(dbType);
       const {
+        AmplifySQLLayerNotificationTopicAccount,
+        AmplifySQLLayerNotificationTopicName,
+        SQLLambdaDataSourceLogicalID,
         SQLLambdaIAMRoleLogicalID,
-        SQLPatchingLambdaIAMRoleLogicalID,
         SQLLambdaLogicalID,
+        SQLPatchingLambdaIAMRoleLogicalID,
         SQLPatchingLambdaLogicalID,
-        SQLLambdaDataSourceLogicalID,
         SQLPatchingSubscriptionLogicalID,
+        SQLPatchingTopicLogicalID,
+        SQLStackName,
       } = ResourceConstants.RESOURCES;
-      const lambdaRoleScope = context.stackManager.getScopeFor(SQLLambdaIAMRoleLogicalID, SQL_STACK_NAME);
-      const lambdaScope = context.stackManager.getScopeFor(SQLLambdaLogicalID, SQL_STACK_NAME);
-      setRDSLayerMappings(lambdaScope, context.rdsLayerMapping);
+      const lambdaRoleScope = context.stackManager.getScopeFor(SQLLambdaIAMRoleLogicalID, SQLStackName);
+      const lambdaScope = context.stackManager.getScopeFor(SQLLambdaLogicalID, SQLStackName);
+
+      // Bimodal behavior alert: In the Gen1 CLI flow, the transform-graphql-schema-v2 buildAPIProject function retrieves the latest layer
+      // version from the S3 bucket. In the CDK construct, such async behavior at synth time is forbidden, so we use an AwsCustomResource to
+      // resolve the latest layer version. The AwsCustomResource does not work with the CLI custom synth functionality, so we fork the
+      // behavior at this point.
+      let layerVersionArn: string;
+      if (context.rdsLayerMapping) {
+        setRDSLayerMappings(lambdaScope, context.rdsLayerMapping);
+        layerVersionArn = Fn.findInMap(ResourceConstants.RESOURCES.SQLLayerMappingID, Fn.ref('AWS::Region'), 'layerRegion');
+      } else {
+        const layerVersionCustomResource = createLayerVersionCustomResource(lambdaScope);
+        layerVersionArn = layerVersionCustomResource.getResponseField('Body');
+      }
```

Comment:
N.p. but I'd prefer to extract the modal behavior, and let/secondary assignment into a separate function, that'll also give you a natural place for this comment.

Reply:
Will refactor in the likely followup to fix e2e tests.